services/bepro-service.ts

```diff
@@ -1,301 +1,413 @@
+    const erc20 = await this.getERC20Obj(tokenAddress);
 
-      await customNetwork.loadContract();
+    return erc20.approve(this.network.contractAddress, amount);
+  }
 
-      return customNetwork;
+  async isApprovedSettlerToken() {
+    if (this.isStarted) {
+      const settler = this.network.settlerToken;
+
+      return settler.isApproved(this.network.contractAddress, 1);
     }
 
-    return this.network;
+    return false;
   }
 
-  async getClosedIssues(networkAddress = undefined) {
+  async isNetworkAbleToClose(networkAddress = undefined) {
     const network = await this.getNetworkObj(networkAddress);
 
-    return network.getAmountOfIssuesClosed();
+    const totalSettlerLocked = await network.totalSettlerLocked();
+    const closedBounties = await network.closedBounties();
+    const canceledBounties = await network.canceledBounties();
+    const bountiesTotal = await network.bountiesIndex();
+
+    return (
+      totalSettlerLocked === 0 &&
+      closedBounties + canceledBounties === bountiesTotal
+    );
   }
 
-  async getSettlerTokenName(networkAddress = undefined) {
-    const network = await this.getNetworkObj(networkAddress);
+  async isCouncil(address = this.address) {
+    if (this.isStarted) {
+      const councilAmount = await this.getNetworkParameter("councilAmount");
+      const oraclesOf = await this.getOraclesOf(address);
 
-    return network.settlerToken.name();
+      return oraclesOf >= councilAmount;
+    }
+
+    return false;
   }
 
-  async getTransactionalTokenName(networkAddress = undefined) {
-    const network = await this.getNetworkObj(networkAddress);
+  async claimNetworkGovernor(networkAddress) {
+    const network = new Network_v2(this.bepro, networkAddress);
+
+    await network.loadContract();
 
-    return network.transactionToken.name();
+    return network.sendTx(network.contract.methods.claimGovernor());
+  }
+  
+  // TODO isApprovedTransactionalToken
+  async isApprovedTransactionalToken(): Promise<boolean> {
+    return false;
   }
 
-  async getOpenIssues(networkAddress = undefined) {
-    const network = await this.getNetworkObj(networkAddress);
+  
+  // TODO closeNetwork
+  // TODO createNetwork
 
-    const quantity = await network.getAmountOfIssuesOpened();
 
-    return quantity - 1;
-  }
 
-  async getBeproLocked(networkAddress = undefined) {
-    const network = await this.getNetworkObj(networkAddress);
+  // Getters and Setters
+  async getNetworkParameter(parameter: "councilAmount" | 
+    "disputableTime" | 
+    "draftTime" | 
+    "oracleExchangeRate" | 
+    "mergeCreatorFeeShare" | 
+    "percentageNeededForDispute") {
+    if (this.isStarted) return this.network[parameter]();
 
-    return network.getBEPROStaked();
+    return 0;
   }
 
-  async getTokensStaked(networkAddress = undefined) {
-    const network = await this.getNetworkObj(networkAddress);
+  async setNetworkParameter(parameter: "councilAmount" | 
+  "disputableTime" | 
```

Comment:
missing spaces will confuse readers, easier to define a type with all these values,

```
type NetworkParameter: "councilAmount" | ...

async setNetworkParameter(param: NetworkParameter) {}
```